libraries/ms-common/inc/spi.h

```diff
@@ -0,0 +1,80 @@
+// HAL SPI
+
+#pragma once
+
+#include <stdint.h>
+
+#include "gpio.h"
+#include "status.h"
+
+// For setting clock polarity
+typedef enum {
+  SPI_CPOLARITY_LOW = 0,
+  SPI_CPOLARITY_HIGH,
+} SPICpol;
+
+// For setting clock phase
+typedef enum {
+  SPI_CPHASE_1EDGE = 0,
+  SPI_CPHASE_2EDGE,
+} SPICpha;
+
+// For setting baud rate prescaler
+typedef enum {
+  SPI_BAUDRATE_2 = 0,
+  SPI_BAUDRATE_4,
+  SPI_BAUDRATE_8,
+  SPI_BAUDRATE_16,
+  SPI_BAUDRATE_32,
+  SPI_BAUDRATE_64,
+  SPI_BAUDRATE_128,
+  SPI_BAUDRATE_256,
+} SPIBaudRate;
```

Comment:
I think it'd be nicer if we could just pass in a desired clock speed and the clock was set up properly.